### PR TITLE
Simple token approval UI

### PR DESCRIPTION
**Pull request by KillariDev**
fixes https://github.com/DarkFlorist/TheInterceptor/issues/190

image

### CODE REVIEW

**Review by jubalm — APPROVED** [review]

File: extension/app/ts/components/simulationExplaining/customExplainers/SimpleTokenApprovalVisualisation.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import { BigAddress } from '../../subcomponents/address.js'
+import { Token721AmountField, TokenAmount, TokenSymbol } from '../../subcomponents/coins.js'
+import { GasFee } from '../SimulationSummary.js'
+
+export function SimpleTokenApprovalVisualisation({ transaction, renameAddressCallBack }: { transaction: SimulatedAndVisualizedTransaction, renameAddressCallBack: RenameAddressCallBack }) {
+	if (transaction.to === undefined) throw new Error('Contract creation trasaction')
+	if (transaction.tokenResults.length != 1) throw new Error('Multiple token events transfer')
+
+	const approval = transaction.tokenResults[0]
+	if (!approval.isApproval) throw new Error('Is Not Approval')
```
Comment (MicahZoltu) [inline extension/app/ts/components/simulationExplaining/customExplainers/SimpleTokenApprovalVisualisation.tsx:12]: Where is this error caught?  Is it presented?

**Review by MicahZoltu — APPROVED** [review]
Would like to better understand error handling.

File: extension/app/ts/components/simulationExplaining/customExplainers/SimpleTokenApprovalVisualisation.tsx (hunk 1 above)
Comment (KillariDev) [reply extension/app/ts/components/simulationExplaining/customExplainers/SimpleTokenApprovalVisualisation.tsx:12]: this should not never occur as its tested beforehand, so its not caught anywhere. If its thrown, something has wrong elsewhere. I guess one option would be to add it to the typing so we wouldn't need to check it

Comment (MicahZoltu) [reply extension/app/ts/components/simulationExplaining/customExplainers/SimpleTokenApprovalVisualisation.tsx:12]: There are several errors being thrown, are these all critical programmer errors and not something we should reasonably be able to deal with?

Comment (KillariDev) [reply extension/app/ts/components/simulationExplaining/customExplainers/SimpleTokenApprovalVisualisation.tsx:12]: which can be fixed with typings, but I was lazy :D

Comment (MicahZoltu) [reply extension/app/ts/components/simulationExplaining/customExplainers/SimpleTokenApprovalVisualisation.tsx:12]: Recommending fixing with type checking then.  😀

**Pull request merged**